sql/core/src/main/scala/org/apache/spark/sql/execution/AggregatingAccumulator.scala

```diff
@@ -205,6 +213,15 @@ class AggregatingAccumulator private(
     buffer = other.buffer
     joinedRow = other.joinedRow
   }
+
+  private[execution] def addState(other: AggregatingAccumulator): Unit = {
+    if (isZero) {
```

Comment:
Why do we need this logic? Why not call merge in `CollectMetricExec`?

Reply:
As I commented [here](https://github.com/apache/spark/pull/32786#discussion_r647007683), `merge` is not intended to be called directly from executor.

Reply:
IMO this method does not really add value, it will just be confusing for whomever comes after this and tries to understand why it is needed. While you are right that the documentation for the accumulator says that merge for an accumulator is not intended to be called on the executor, you are allowed to widen the scope of a method in a subclass as it won't violate the contract.